src/materialized/src/lib.rs

```diff
@@ -247,21 +265,22 @@ pub async fn serve(config: Config) -> Result<Server, anyhow::Error> {
         experimental_mode: config.experimental_mode,
         now: SYSTEM_TIME.clone(),
         metrics_registry: config.metrics_registry.clone(),
+        persister: persister.runtime.clone(),
     })?;
 
     // Initialize coordinator.
     let (coord_handle, coord_client) = coord::serve(coord::Config {
         dataflow_client: Box::new(dataflow_client),
         logging: config.logging,
-        data_directory: &config.data_directory,
+        storage: coord_storage,
```

Comment:
I realized this is just from movement but it's a little confusing for this to be called storage when now we have STORAGE. IIUC this is more akin to the stash you're introducing in #10306, so maybe call it `stash`?

Reply:
```
$ git grep "storage" -- src/coord | wc -l
      52
```

nevermind. if we do this, it should be its own PR